helix-core/src/test/java/org/apache/helix/controller/changedetector/TestResourceChangeDetector.java

```diff
@@ -438,7 +441,7 @@ public void testResetSnapshots() {
     Assert.assertEquals(
         changeDetector.getAdditionsByType(ChangeType.IDEAL_STATE).size() + changeDetector
             .getChangesByType(ChangeType.IDEAL_STATE).size() + changeDetector
-            .getRemovalsByType(ChangeType.IDEAL_STATE).size(), 2);
+            .getRemovalsByType(ChangeType.IDEAL_STATE).size(), 1);
```

Comment:
Does this number change mean what we previously tested was wrong? How to justify?

Reply:
see line 418, 419
```
      // remove newly added resource/ideastate
      _gSetupTool.getClusterManagementTool().dropResource(CLUSTER_NAME, resourceName);
```

The newly added resource in the previous test is not really valid. Or they would break this test. So in this diff, it is removed.  Accordingly the number here needs to be adjusted too.